### DISCUSSION TITLE
Add a "zx input area" for touch screen players / 为触摸屏玩家增加一个zx输入区

### DISCUSSION THREAD

**Discussion by ReBiliBin** [ppy/osu#27725] [Ideas]
_**Google translated English ver.**_
(i'm a chinese and my english is so poor , soi'm using translator , please forgive me...)
(Original Chinese ver is in the bottom of the English ver.)

### REASON

I discovered while playing osu using a touch device (my Xiaomi Pad5) that sometimes the cursor would be positioned on the part that was being tapped as zx input. This will result in an inability to hit notes, eventually leading to combo break, or even game failure. So I was wondering if it would improve the situation if a dedicated zx input section could be added somewhere in the game specifically for touch screen players.

**_Chinese ver._**

### 原因

我发现我在用我的触摸设备（小米平板5）打osu时，光标有时会跑到我用来作为zx输入的手下，导致断连甚至GAMEOVER。所以我在想，可不可以在某处添加一个给触摸玩家使用的 专门用来输入zx的区域，防止这种情况（至少改善）这种情况的发生。

**Comment by ReBiliBin** [comment]
[lijiale2217 playing cYsmix feat. Emmy - Tear Rain (jonathanlfj) [Insane] (2024-03-25_20-28).zip](https://github.com/ppy/osu/files/14743573/lijiale2217.playing.cYsmix.feat.Emmy.-.Tear.Rain.jonathanlfj.Insane.2024-03-25_20-28.zip)
I successfully reproduced this problem. Here is my replay(.osr) file.

**Comment by Walavouchey** [comment]
you can use the screen scaling options to move the playfield out of the way. tapping outside the playfield will always register as a tap and not as a cursor input

![osu_2024-03-25_14-02-39](https://github.com/ppy/osu/assets/36758269/d3900f58-72ca-4e6a-a8a2-c9554a1f9233)

the idea of dedicated tapping areas has been mentioned before, but i can't find an existing issue for it...

**Reply by shiumano** [reply]
There is no such thing
It simply moves the cursor outside the playfield.
(I also expected this to happen when I saw the update announcement, but it didn't happen. I'm experiencing the same pain.)

**Reply by Walavouchey** [reply]
allow me to correct/clarify myself:

tapping on a note will always update the cursor position, regardless of whether you were already holding one finger somewhere else. by tapping outside the playfield, you're guaranteed not to hit a note, which means those taps never register as cursor inputs while you're holding another finger down

moving the playfield via screen scaling options helps increase the "safe area" so you don't accidentally hit a note with your tapping hand

**Reply by shiumano** [reply]
Touch input implemented PR: #22375

The problem is when the following flow occurs
The touch input of the person being tracked is interrupted by a glitch
↓
At that moment, ZX input is made
↓
Tracked touch moves to the ZX side
↓
The originally tracked touch input becomes helpless unless you are lucky enough to step on the hit circle due to jitter.

Only Direct touch can be avoided by touching outside the playfield.
[2024年3月28日 [NMrktP1HLUo].webm](https://github.com/ppy/osu/assets/75152752/4d557c11-24cf-4521-97eb-e6f486b4374f)

**Reply by ThatStella7922** [reply]
Noticed this too, in short the issue is

1. You are holding with one finger over notes on the playfield
2. You are tapping with two other fingers, **not** over notes
3. If a note happens to come under the fingers you are tapping with, and you tap the note, you will combo break
4. You have to let go of the first finger that you were holding with and place it back down 
5. If you do it fast enough you can avoid failing

I think the expected resolution would probably be to emulate the functionality of osu!droid, where step 3 did not happen
